### PR TITLE
Added role info when register Swarm to Mesos.

### PR DESCRIPTION
**Pull request by klaus1982**
Signed-off-by: Klaus Ma <klaus1982.cn@gmail.com>

### CODE REVIEW

File: cluster/mesos/cluster.go [hunk 1]
```diff
@@ -82,13 +83,19 @@ func NewCluster(scheduler *scheduler.Scheduler, TLSConfig *tls.Config, master st
 	// Empty string is accepted by the scheduler.
 	user, _ := options.String("mesos.user", "SWARM_MESOS_USER")
 
+	var found bool
+
+	if cluster.role, found = options.String("mesos.role", "SWARM_MESOS_ROLE"); !found {
```
Comment (jimenez) [inline cluster/mesos/cluster.go]: What do you think if we put our `var found bool` inlined with the `if` as follow: 
```
if role, found := options.String("mesos.role", "SWARM_MESOS_ROLE"); found {
 cluster.role = role
}
``` 

And make the default `"*"` part of the initialization of Cluster line 78.

**Comment by jimenez** [comment]
Thanks @klaus1982 ! Could you also add an integration test for this?

**Comment by klaus1982** [comment]
Sure, I'll add some integration test for it.

**Comment by jimenez** [comment]
Hi @klaus1982 , it'd be great if you could rebase this :)

**Comment by vieux** [comment]
@klaus1982 please ping use to reopen once you rebased.

**Pull request closed without merging**